monai/networks/nets/efficientnet.py

```diff
@@ -785,18 +799,6 @@ def _decode_block_list(string_list: List[str]):
     Returns:
         blocks_args: a list of BlockArgs namedtuples of block args.
     """
-    # Parameters for an individual model block
-    # namedtuple with defaults for mypy help from:
-    # https://stackoverflow.com/a/53255358
-    class BlockArgs(NamedTuple):
-        num_repeat: int
-        kernel_size: int
-        stride: int
-        expand_ratio: int
-        input_filters: int
-        output_filters: int
-        id_skip: bool
-        se_ratio: Optional[float] = None
 
     def _decode_block_string(block_string: str):
```

Comment:
looks like the `_decode_block_string` should be a part of `BlockArgs`, and we could add a `to_string` method for completeness?

```py
class BlockArgs(NamedTuple):
    ...
    @staticmethod
    def from_string(inst_string):
        ...
        return a blockargs instance

    def to_string(self):
        ...
        # convert args into the string notation

# usage:
blocks_args = [BlockArgs.from_string(s) for s in string_list]
strings = [b.to_string() for b in block_args]
```

Reply:
@wyli yes, it can be done like that. Let me correct it that way. The reason I left this mechanism of decoding model definitions from a string was to make it future proof where in future we can have support for new EfficientNet model definitions learned through AutoML. These are usually defined as list of strings that can be decoded to make the model.

Reply:
exactly, it's a nice idea, let's revisit this when we make the generic AutoML-related APIs